### DISCUSSION TITLE
Compilation succeeded but there are warning(

### DISCUSSION THREAD

**Discussion by roman-int3** [quasarframework/quasar#10471] [General - Components / Directives / etc]
Hello
Can anybody point to me where is the problem with quasar build process?
I receive this warning and after app is start it crash with error

**build warning:**

App •  WARNING  •  UI  in ./node_modules/pouchdb-utils/node_modules/uuid/wrapper.mjs
export 'default' (imported as 'uuid') was not found in './dist/index.js' (possible exports: __esModule, v1, v3, v4, v5)
 App •  WARNING  • Compilation succeeded but there are warning(s). Please check the log above.
 App • Opening default browser at http://localhost:8080

**app error in browser:**

[Quasar] Running SPA.
wrapper.mjs?a4aa:2 Uncaught (in promise) TypeError: Cannot read property 'v1' of undefined
    at eval (webpack-internal:///./node_modules/pouchdb-utils/node_modules/uuid/wrapper.mjs:10)
    at Module../node_modules/pouchdb-utils/node_modules/uuid/wrapper.mjs (vendor.js:7786)
    at __webpack_require__ (app.js:361)
    at fn (app.js:626)
    at eval (webpack-internal:///./node_modules/pouchdb-utils/lib/index-browser.es.js:40)
    at Module../node_modules/pouchdb-utils/lib/index-browser.es.js (vendor.js:5207)
    at __webpack_require__ (app.js:361)
    at fn (app.js:626)
    at eval (webpack-internal:///./node_modules/rxdb/dist/es/util.js:43)
    at Module../node_modules/rxdb/dist/es/util.js (vendor.js:6191)

**Comment by MatthewSH** [comment]
Are you using `uuid` in your code anywhere? I don't think Quasar uses it, but it may be required by some dependency somewhere. If you are using it in your codebase, have you Googled it? I figured based on the error that you were importing the module incorrectly, but instead of assuming...I just Googled it and it led me to [this SO post](https://stackoverflow.com/questions/60830848/attempted-import-error-uuid-does-not-contain-a-default-export-imported-as-u) as the first result.

Really I just wanna make sure you or whatever you're using it leveraging the library according to their [docs](https://github.com/uuidjs/uuid#quickstart)

Based on the error it seems like you're trying to import and use it like:
```javascript
import uuid from 'uuid';

var someuuid = uuid.v1()
```
or something like it. There's no default export in `uuid`, so that style of importing is not valid and can cause errors like that.

**Reply by roman-int3** [reply]
uuid used by pouchdb-utils, and all works fine in vuetify project now i transfer project to quasar and recieved this error, i think problem in webpack config, pounchdb-utils contains uuid as nested dependency in /node_modules/pouchdb-utils/node_modules/uuid/ and  in this folder i have wrapper.mjs file, when i remove him all build is sussesfull and app work correct.

**Reply by MatthewSH** [reply]
Are you creating a new quasar project or trying to implement Quasar into an existing project?
Also, in your `package-lock` or `yarn.lock` could you share the `uuid` version in it?

**Reply by roman-int3** [reply]
this is new quasar project.
i created test project, please check

https://github.com/roman-int3/app

src/boot/database.ts

**Reply by MatthewSH** [reply]
So this is not a Quasar problem, this is a PouchDB problem. They lock it at uuid@8.1.0 (in [pouchdb@7.2.2](https://github.com/pouchdb/pouchdb/blob/7.2.2/package.json#L47)) where this problem exists (https://github.com/uuidjs/uuid/issues/561, https://github.com/uuidjs/uuid/pull/563). On [pouchdb master](https://github.com/pouchdb/pouchdb/blob/master/package.json#L61) it's updated to the latest version.

Until they publish a new version of the libraries, there's not much that can be done that wouldn't be hacky at best. I worked around it by modifying the lock files and manually assigning the versions, but that's only for me testing my suspicion (which turned out to be true) and you definitely shouldn't do that on a production application as that file is updated by NPM.

**Reply by roman-int3** [reply]
Thanks mate, for your help.
